src/main/resources/public/ts/sniplets/workspace-nextcloud-folder.sniplet.ts

```diff
@@ -167,34 +190,37 @@ class ViewModel implements IViewModel {
 
     setSwitchDisplayHandler(): void {
         const viewModel: IViewModel = this;
-        const $workspaceFolderTree: JQuery = $(WorkspaceEntcoreUtils.$ENTCORE_WORKSPACE);
+        const $workspaceFolderTree: JQuery = $(WorkspaceEntcoreUtils.$ENTCORE_WORKSPACE + ' li > a');
         // case nextcloud folder tree is interacted
         // checking if listener does not exist in order to create one
-        if (!(<any>$)._data($($nextcloudFolder)[0], "events").click) {
-            $($nextcloudFolder).click(this.switchWorkspaceTreeHandler());
-        }
+        $($nextcloudFolder)
+            .off('click.workspaceNextcloudHandler')
+            .on('click.workspaceNextcloudHandler', this.switchWorkspaceTreeHandler());
 
         // case entcore workspace folder tree is interacted
-        // checking if listener does not exist in order to create one
-        if (!(<any>$)._data($workspaceFolderTree.get(0), "events")) {
-            $workspaceFolderTree.click(this.switchNextcloudTreeHandler(viewModel));
-        }
+        // we unbind its handler and rebind it in order to keep our list of workspace updated
+        $workspaceFolderTree
+            .off('click.nextcloudHandler')
+            .on('click.nextcloudHandler', this.switchNextcloudTreeHandler(viewModel));
     }
 
     /**
      * Remove workspace tree and use nextcloud tree instead.
      */
     private switchWorkspaceTreeHandler() {
+        const viewModel: IViewModel = this;
         return function (): void {
+            if (!viewModel.selectedFolder) {
+                viewModel.folderTree.openFolder(viewModel.documents[0]);
```

Comment:
viewModel.documents[0] sera toujours initialisé car c'est le this.document qu'on "init" dans le constructeur